fix(Listbox): rtl mode

PurwaShrivastava opened the pull request:
<!--
Please make sure to read the Pull Request Guidelines:
https://github.com/qlik-oss/nebula.js/blob/master/.github/CONTRIBUTING.md#git
-->

## Motivation

RTL mode for the ListBox. This is to fix https://github.com/qlik-oss/sn-list-objects/issues/82

Before: Title, Search Icon, Columns & Toolbar were not correctly aligned for Folded as well as normal ListBox.
Number Alignment was not correct & - sign for negative numbers shown backwards.

After: Attached Screenshots
Litbox Title, Search Glass & Columns with Numeric positive negative as well as String values in RTL mode :

![Screenshot 2023-03-15 at 11 36 50](https://user-images.githubusercontent.com/30860406/225583353-ba40b92f-da32-40eb-8d71-9a97d846dc4a.png)

Actions Toolbar in RTL Mode :

![Screenshot 2023-03-16 at 11 06 18](https://user-images.githubusercontent.com/30860406/225584139-99d2e742-f51a-42d1-84c4-e7632d48d0fe.png)

Folded ListBox in RTL Mode : 

![Screenshot 2023-03-15 at 11 37 20](https://user-images.githubusercontent.com/30860406/225583193-8e1eabb3-fa98-48b1-90ba-f95f2d3cbffa.png)

## Requirements checklist

<!-- Make sure you got these covered -->

- [x] Api specification
  - [ ] Ran `yarn spec`
    - [ ] No changes **_OR_** API changes has been formally approved
- [x] Unit/Component test coverage
- [x] Correct PR title for the changes (fix, chore, feat)

When build and tests have passed:

- [ ] Add code reviewers, for example @qlik-oss/nebula-core

Caele commented on apis/nucleus/src/components/listbox/ListBoxInline.jsx:
```diff
@@ -301,6 +319,7 @@ function ListBoxInline({ options, layout }) {
                 onConfirm: () => {},
                 onCancel: () => {},
               }}
+              direction={isRtl}
```
This doesn't look right? 
Could we use `direction` (rtl/ltr) for all properties passed between components and only use `isRtl` for checks witin a component? Its easier to understand if we are consistent in what properties mean.

PurwaShrivastava replied:
ok shall change

Caele commented on apis/nucleus/src/components/ActionsToolbar.jsx:
```diff
@@ -45,9 +45,9 @@ const ActionToolbarElement = {
 
 const ActionsGroup = React.forwardRef(
-  ({ ariaExpanded = false, actions = [], first = false, last = false, addAnchor = false }, ref) =>
+  ({ ariaExpanded = false, actions = [], first = false, last = false, addAnchor = false, isRtl = false }, ref) =>
```
This using isRtl as a property here.

PurwaShrivastava replied:
Have forwarded direction in the main element ActionsToolbar, calculated isRTL and using it inside the inner element, seems redundant to pass direction again and recalculate isRtl

Caele commented:
- Look at adding a rendering test to cover the RTL case
- The current rendering tests should not change or?

Caele commented:
> I have updated the snapshots for rendering tests, have added unit tests for RTL case. Can look into adding rendering tests.

I meant that the changes you have made should not affect the current rendering tests, so was there an intentional style update there somewhere or?

PurwaShrivastava commented:
> > I have updated the snapshots for rendering tests, have added unit tests for RTL case. Can look into adding rendering tests.
> 
> I meant that the changes you have made should not affect the current rendering tests, so was there an intentional style update there somewhere or?



> > I have updated the snapshots for rendering tests, have added unit tests for RTL case. Can look into adding rendering tests.
> 
> I meant that the changes you have made should not affect the current rendering tests, so was there an intentional style update there somewhere or?

Have introduced a right padding for header.
